sites/all/modules/reol_app_feeds/src/Helper/ParagraphHelper.php

```diff
@@ -0,0 +1,1001 @@
+   *
+   * @return array
+   *   The theme list data.
+   */
+  private function getThemeList(\ParagraphsItemEntity $paragraph) {
+    // Get all items.
+    //
+    // Note: We may get more data than actually needed, but due to non-trivial
+    // filtering on "identifiers" (see below) we have to load all items.
+    //
+    // @TODO: Can we improve this so we don't have to load all items and throw
+    // away some of them?
+    $items = $this->nodeHelper->loadReferences($paragraph, 'field_picked_articles');
+
+    $list = array_values(array_map([$this, 'getThemeData'], $items));
+    // Remove items with no identifiers.
+    $list = array_values(array_filter($list, function (array $item) {
+      return isset($item['identifiers']);
+    }));
+
+    // Slice the list to the maximum allowed length.
+    $list = array_slice($list, 0, (int) _reol_app_feeds_variable_get('reol_app_feeds_frontpage', 'theme_list_max_length', 6));
+
+    return [
+      'guid' => $this->getGuid($paragraph),
+      'type' => $this->getType($paragraph),
+      'view' => $this->getView($paragraph),
+      'list' => $list,
+    ];
+  }
+
+  /**
+   * Get theme data.
+   *
+   * @param object $node
+   *   The node.
+   *
+   * @see https://docs.google.com/document/d/1lJ3VPAJf7DAbBWAQclRHfcltzZefUG3iGCec-z97KlA/edit?ts=5c4ef9d5#bookmark=id.a1elwnwq3nk4
+   *
+   * @return array
+   *   The theme data.
+   */
+  public function getThemeData($node) {
+    $view = $this->nodeHelper->getFieldValue($node, 'field_image_teaser', 'value') ? 'image' : 'covers';
+    $contentType = $this->nodeHelper->getFieldValue($node, 'field_article_type', 'value');
+    $type = $this->nodeHelper->getThemeType($contentType);
+
+    $image = !empty($node->field_ding_news_list_image) ? $this->nodeHelper->getImage($node->field_ding_news_list_image, FALSE, 'app_news_image') : static::VALUE_NONE;
+    $identifiers = $this->nodeHelper->getTingIdentifiers($node, 'field_ding_news_materials');
+
+    // Hack for eReolen Go!
+    if ('breol_news' === $node->type) {
+      $view = 'image';
+      $image = !empty($node->field_breol_cover_image) ? $this->nodeHelper->getImage($node->field_breol_cover_image) : static::VALUE_NONE;
+
+      // Get identifiers from carousel queries.
+      $carousels = $this->nodeHelper->getFieldValue($node, 'field_carousels', NULL, TRUE);
+      $identifiers = [];
+      foreach ($carousels as $carousel) {
+        module_load_include('inc', 'opensearch', 'opensearch.client');
+        $result = opensearch_do_search($carousel['search']);
+        foreach ($result->collections as $collection) {
+          /** @var \TingCollection $collection */
+          /** @var \TingEntity $object */
+          $object = $collection->getPrimary_object();
+          $identifier = $object->getId();
+          if (!in_array($identifier, $identifiers)) {
+            $identifiers[] = $identifier;
+          }
+        }
+        // Only get identifiers from the first query.
+        break;
```

Comment:
If loop is always stop here no matter the content of `$identifiers` why not drop the loop an just use:

```
$carousel = reset($carousels)
```

Reply:
We're inside a hack block!

Reply:
And... you do a `foreach ($carousels as $carousel)` and always just look at the first one?

Reply:
Yes, we're inside a hack block! The customer may change his mind.

Reply:
Okay, that way around...